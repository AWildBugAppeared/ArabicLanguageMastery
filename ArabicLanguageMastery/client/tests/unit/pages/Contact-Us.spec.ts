/* eslint-disable  @typescript-eslint/no-explicit-any */

// Imports
import Vue from 'vue';
import Vuetify from 'vuetify';

// Utilities
import { createLocalVue, mount, Wrapper } from '@vue/test-utils';

import ContactUs from '@/pages/Contact-Us.vue';

describe('ContactUs.vue', () => {
  const localVue = createLocalVue();
  let wrapper: Wrapper<Vue & { [key: string]: any }>;
  let vuetify: Vuetify;

  const mountWrapper = () => {
    wrapper = mount(ContactUs, {
      localVue,
      vuetify,
    });
  };

  beforeEach(() => {
    vuetify = new Vuetify({
      mocks: {
        $vuetify: {},
      },
    });
    Vue.use(Vuetify);
    document.body.setAttribute('data-app', 'true');
  });

  afterEach(() => {
    wrapper.destroy();
    jest.resetModules();
    jest.clearAllMocks();
  });

  it('should display contact us form', async () => {
    mountWrapper();
    await wrapper.vm.$nextTick();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
