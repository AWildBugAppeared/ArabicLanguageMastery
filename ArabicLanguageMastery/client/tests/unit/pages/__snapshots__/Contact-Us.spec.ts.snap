// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUs.vue should display contact us form 1`] = `
"<div class=\\"container text-center\\">
  <form name=\\"contact\\" method=\\"POST\\" data-netlify=\\"true\\" data-netlify-honeypot=\\"bot-field\\"><input type=\\"hidden\\" value=\\"contact\\" name=\\"form-name\\">
    <div class=\\"v-input theme--light v-text-field\\">
      <div class=\\"v-input__control\\">
        <div class=\\"v-input__slot\\">
          <div class=\\"v-text-field__slot\\"><label for=\\"input-2\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Your Name</label><input name=\\"name\\" id=\\"input-2\\" type=\\"text\\"></div>
        </div>
        <div class=\\"v-text-field__details\\">
          <div class=\\"v-messages theme--light\\">
            <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
          </div>
        </div>
      </div>
    </div>
    <div class=\\"v-input theme--light v-text-field\\">
      <div class=\\"v-input__control\\">
        <div class=\\"v-input__slot\\">
          <div class=\\"v-text-field__slot\\"><label for=\\"input-5\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Your Email</label><input name=\\"email\\" id=\\"input-5\\" type=\\"email\\"></div>
        </div>
        <div class=\\"v-text-field__details\\">
          <div class=\\"v-messages theme--light\\">
            <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
          </div>
        </div>
      </div>
    </div>
    <div class=\\"v-input v-textarea theme--light v-text-field\\">
      <div class=\\"v-input__control\\">
        <div class=\\"v-input__slot\\">
          <div class=\\"v-text-field__slot\\"><label for=\\"input-8\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Message</label><textarea name=\\"message\\" id=\\"input-8\\" rows=\\"5\\"></textarea></div>
        </div>
        <div class=\\"v-text-field__details\\">
          <div class=\\"v-messages theme--light\\">
            <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
          </div>
        </div>
      </div>
    </div> <button type=\\"submit\\" class=\\"v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default\\"><span class=\\"v-btn__content\\">Send</span></button>
  </form>
</div>"
`;
